controllers/mongodb_cleanup.go

```diff
@@ -2,9 +2,27 @@ package controllers
 
 import (
 	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/constants"
 	"k8s.io/apimachinery/pkg/types"
 )
 
+// cleanupPemSecret cleans up the old pem secret generated for the agent certificate.
+func (r *ReplicaSetReconciler) cleanupPemSecret(currentMDB mdbv1.MongoDBCommunitySpec, lastAppliedMDBSpec mdbv1.MongoDBCommunitySpec, namespace string) {
+	if currentMDB.GetAgentAuthMode() == lastAppliedMDBSpec.GetAgentAuthMode() {
+		return
+	}
+
+	if !currentMDB.IsAgentX509() && lastAppliedMDBSpec.IsAgentX509() {
+		agentCertSecret := lastAppliedMDBSpec.GetAgentCertificateRef()
+		if err := r.client.DeleteSecret(types.NamespacedName{
+			Namespace: namespace,
+			Name:      agentCertSecret + "-pem",
+		}); err != nil {
```

Comment:
nit: we can check for not found and in that case only log a debug info. SInce that just means something already did our work

Reply:
Done!